File: Examples/test-suite/octave/catches_strings_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

catches_strings

exception_thrown = false;
try
  StringsThrower.charstring();
catch e
  if (isempty(strfind(e.message, "charstring message")))
    error("incorrect exception message: %s", e.message)
  endif
  exception_thrown = true;
end_try_catch
if (!exception_thrown)
  error("Should have thrown an exception");
endif

exception_thrown = false;
try
  StringsThrower.stdstring();
catch e
  if (isempty(strfind(e.message, "stdstring message")))
    error("incorrect exception message: %s", e.message)
  endif
  exception_thrown = true;
end_try_catch
if (!exception_thrown)
  error("Should have thrown an exception");
endif

File: Examples/test-suite/php/catches_strings_runme.php
<?php

require "tests.php";

$exception_thrown = false;
try {
  StringsThrower::charstring();
} catch (Exception $e) {
  check::str_contains($e->getMessage(), "charstring message", "incorrect exception message: {$e->getMessage()}");
  $exception_thrown = true;
}
check::equal($exception_thrown, true, "Should have thrown an exception");

$exception_thrown = false;
try {
  StringsThrower::stdstring();
} catch (Exception $e) {
  check::str_contains($e->getMessage(), "stdstring message", "incorrect exception message: {$e->getMessage()}");
  $exception_thrown = true;
}
check::equal($exception_thrown, true, "Should have thrown an exception");

check::done();
